circuits/src/zk_circuits/valid_relayer_fee_settlement.rs

```diff
@@ -415,3 +415,589 @@ where
         Self::circuit(&statement_var, &witness_var, cs).map_err(PlonkError::CircuitError)
     }
 }
+
+#[cfg(any(test, feature = "test_helpers"))]
+pub mod test_helpers {
+    //! Helper methods for testing the `VALID RELAYER FEE SETTLEMENT` circuit
+
+    use circuit_types::{
+        keychain::DecryptionKey,
+        native_helpers::{
+            compute_wallet_private_share_commitment, compute_wallet_share_commitment,
+            compute_wallet_share_nullifier, reblind_wallet,
+        },
+        wallet::Wallet,
+        Amount,
+    };
+    use rand::{thread_rng, Rng, RngCore};
+
+    use crate::zk_circuits::test_helpers::{create_multi_opening, create_wallet_shares};
+
+    use super::{ValidRelayerFeeSettlementStatement, ValidRelayerFeeSettlementWitness};
+
+    /// Create a valid witness and statement for the `VALID RELAYER FEE
+    /// SETTLEMENT` circuit
+    pub fn create_witness_statement<
+        const MAX_BALANCES: usize,
+        const MAX_ORDERS: usize,
+        const MERKLE_HEIGHT: usize,
+    >(
+        sender_wallet: &Wallet<MAX_BALANCES, MAX_ORDERS>,
+        recipient_wallet: &Wallet<MAX_BALANCES, MAX_ORDERS>,
+    ) -> (
+        ValidRelayerFeeSettlementStatement<MAX_BALANCES, MAX_ORDERS>,
+        ValidRelayerFeeSettlementWitness<MAX_BALANCES, MAX_ORDERS, MERKLE_HEIGHT>,
+    )
+    where
+        [(); MAX_BALANCES + MAX_ORDERS]: Sized,
+    {
+        let mut rng = thread_rng();
+        let mut sender_wallet = sender_wallet.clone();
+        let mut recipient_wallet = recipient_wallet.clone();
+
+        // Pick balances to send and receive the fees from
+        let send_idx = rng.gen_range(0..MAX_BALANCES);
+        let receive_idx = rng.gen_range(0..MAX_BALANCES);
+        recipient_wallet.balances[receive_idx].mint = sender_wallet.balances[send_idx].mint.clone();
```

Comment:
this assumes that `recipient_wallet` doesn't have a balance w/ the same mint as `sender_wallet.balances[send_idx]`, if it does then the random `receive_idx` may be a different index and we'd violate the invariant of unique balance mints. something the caller should be aware of

Reply:
correct, for testing sake this is okay though because we explicitly mask the index